standalone-metastore/metastore-server/src/test/java/org/apache/hadoop/hive/metastore/TestFilterHooks.java

```diff
@@ -210,6 +225,12 @@ protected void creatEnv(Configuration conf) throws Exception {
 
     client.dropDatabase(DBNAME1, true, true, true);
     client.dropDatabase(DBNAME2, true, true, true);
+    try {
+      client.dropDataConnector(DCNAME1, true, true);
+      client.dropDataConnector(DCNAME2, true, true);
+    } catch (NoSuchObjectException e) {
+      // no op
```

Comment:
I'm wondering why we are doing a no-op for this drop operation since we are already passing true as value for ifnotexists argument. 
So, I don't think we need to catch nosuchobjectexception.

Reply:
When we are dropping dataconnector whch does not exists,  this line will throw NoSuchObjectException though passing ifnotexists argument(true):
https://github.com/apache/hive/blob/cdb1052e24ca493c6486fef3dd8956dde61be834/standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/HMSHandler.java#L2045

If this is a bug, i will fix it in another pr :). thx

Reply:
Yeah, I think this is a bug and needs fixing. Because 
`drop connector if exists unexisting_connector;`
shouldn't throw an error if the connector doesn't exist.

Reply:
https://github.com/apache/hive/pull/3350 I have created a pr to fix this. If you have time, please take a look. :)
cc @saihemanth-cloudera @nrg4878

Reply:
@saihemanth-cloudera Because of  https://github.com/apache/hive/pull/3350 has been merged, i removed the needless catch nosuchobjectexception. Could we continue to merge this pr? Thank you.